bundles/org.openhab.binding.bmwconnecteddrive/README.md

```diff
@@ -0,0 +1,849 @@
+
+Possible states
+
+* _Undef_ - no status data available
+* _Invalid_ - this door / window isn't applicable for this vehicle
+* _Closed_ - the door / window is closed
+* _Open_ - the door / window is open
+* _Intermediate_ - window in intermediate position, not applicable for doors
+
+#### Range Data
+
+Based on vehicle type some channels are present or not. 
+Conventional fuel vehicles don't provide *Electric Range* and battery electric vehicles don't show *Fuel Range*.
+Hybrid vehicles have both and in addition *Hybrid Range*.
+See description [Range vs Range Radius](#range-vs-range-radius) to get more information.
+
+* Channel Group ID is **range**
+* Availability according to table
+* Read-only values
+
+| Channel Label         | Channel ID            | Type                 | conv | phev | bev_rex | bev |
+|-----------------------|-----------------------|----------------------|------|------|---------|-----|
+| Mileage               | mileage               | Number:Length        |  X   |  X   |    X    |  X  |
+| Fuel Range            | range-fuel            | Number:Length        |  X   |  X   |    X    |     |
+| Battery Range         | range-electric        | Number:Length        |      |  X   |    X    |  X  | 
+| Hybrid Range          | range-hybrid          | Number:Length        |      |  X   |    X    |     | 
+| Battery Charge Level  | soc                   | Number:Dimensionless |      |  X   |    X    |  X  |
+| Remaining Fuel        | remaining-fuel        | Number:Volume        |  X   |  X   |    X    |     | 
+| Fuel Range Radius     | range-radius-fuel     | Number:Length        |  X   |  X   |    X    |     | 
+| Electric Range Radius | range-radius-electric | Number:Length        |      |  X   |    X    |  X  | 
+| Hybrid Range Radius   | range-radius-hybrid   | Number:Length        |      |  X   |    X    |     | 
+
+
+#### Charge Profile
+
+Charging options with date and time for preferred time windows and charging modes.
+
+* Channel Group ID is **charge**
+* Available for electric and hybrid vehicles
+* Read/Write access for UI. Use [Charge Profile Actions](#charge-profile-action) in rules
+* There are 3 timers available. Replace *X* with number 1,2 or 3 to target the correct timer
+
+| Channel Label                         | Channel Group ID | Channel ID                | Type     | 
+|---------------------------------------|------------------|---------------------------|----------| 
+| Charging Mode for Profile             | charge           | profile-mode              | String   | 
+| Charging Preferences for Profile      | charge           | profile-prefs             | String   | 
+| Charging Window Start Time            | charge           | window-start              | DateTime | 
+| Charging Window Start Time Hour       | charge           | window-start-hour         | Number   | 
+| Charging Window Start Time Minute     | charge           | window-start-minute       | Number   | 
```

Comment:
*Hour* and *Minute* are available for editing charging profile. I'm thinking now about removing the channel with *DateTimeType* because it holds redundant information.

Reply:
Wouldn't it be more reasonable to keep the DateTime Channel?

Reply:
Generaly speaking I agree on keeping the DateTime-Channel (needs to be included in the command-handling-code though). I did introduce the minute and hour channels because you cannot easily update a dateTime from a simple sitemap (From my understanding you can do using a minute and hour-item plus a rule).
You can do direct updates using a standard OH3-widget. The usability of the stock widget is not really good, but I'd expect this widget can easily be improved independent from this particular usecase.

Reply:
What's your suggestion @ntruchsess ? Shall we remove hour & minute channels?
In OH3 UI you're able to define some options so you can put in e.g. hourly adaptions. Manual text input I didn't see so far

Reply:
I suggest keeping an editable DateTime-channel.
One can do manual text-input in the OH3 UI. Here is an example that combines formated output and manual input:
```
- component: oh-list-card
  config:
    mediaList: true
  slots:
    default:
      - component: oh-list-item
        config:
          subtitle: =dayjs(items.ATestItem.state).format('HH:mm')
          title: ATestItem
          icon: oh:time
      - component: oh-input
        config:
          type: time
          inputmode: text
          placeholder: hh:mm
          sendButton: true
          validate: true
          item: ATestItem
          outline: false

```
the redering of oh-input, type time is platform-dependent. This is not very pretty, but so far it works and I'm pretty sure it can (and will) be improved.

Reply:
please see PR https://github.com/weymann/openhab-addons/pull/13 (simplifies the editable DateTimeType-channel, removes the minute- and hour-based channels

Reply:
I think for the days-channels that also might be simplified using an editable StringListType-Channel instead of 7 OnOffType-channels per week. Implementation in the binding is pretty straightforward, but I do not have a working OH3-UI widget-based example for that yet.